recipes/azure-sdk-for-cpp/all/conanfile.py

```diff
@@ -0,0 +1,108 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout
+from conan.tools.files import get, copy
+from conan.tools.scm import Version
+import os
+
+required_conan_version = ">=1.53.0"
+
+
+class AzureSDKForCppConan(ConanFile):
+    name = "azure-sdk-for-cpp"
+    description = "Microsoft Azure Storage Client Library for C++"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/Azure/azure-sdk-for-cpp"
+    topics = ("azure", "cpp", "cross-platform", "microsoft", "cloud")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    _sdks = (
+        "azure-core",
+        "azure-template",
+        "azure-security-keyvault-administration",
+        "azure-security-keyvault-certificates",
+        "azure-security-keyvault-secrets",
+        "azure-security-attestation",
+        "azure-security-keyvault-keys",
+        "azure-data-tables",
+        "azure-identity",
+        "azure-storage-common",
+        "azure-storage-queues",
+        "azure-storage-files-shares",
+        "azure-storage-blobs",
+        "azure-storage-files-datalake"
+    )
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def configure(self):
+        if self.options.get_safe("shared"):
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        self.requires("libcurl/8.6.0")
+        self.requires("libxml2/2.12.6")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, 14)
+
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration(
+                f"{self.ref} is not supported on {self.settings.os}.")
+
+        if self.settings.compiler != "gcc":
+            raise ConanInvalidConfiguration(
+                f"{self.ref} is not supported on {self.settings.compiler}.")
```

Comment:
By the wording on your PR comment
> and only on Linux.

it seems these limitations are a recipe one, and not something coming from upstream.

If that's the case, I'd suggest changing the wording here to add something like "not supported yet, contributions welcomed!" or something like that :)

Reply:
Clearly that is what i meant. I have updated the comment saying that windows and apple support is not yet donee and contributions welcomes :).

Reply:
I am still not convinced. If I read ``"azure-sdk-for-cpp is not supported on Windows"`` I still think that the library doesn't support Windows. I think it should be more explicit: "The azure-sdk-for-cpp Conan recipe in ConanCenter still does not support Windows, contributions to the recipe welcome"